iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/relational/ColumnTransformerBuilder.java

```diff
@@ -999,6 +1003,57 @@ private ColumnTransformer getFunctionColumnTransformer(
       Type returnType = columnTransformers.get(0).getType();
       return AbstractGreatestLeastColumnTransformer.getLeastColumnTransformer(
           returnType, columnTransformers);
+    } else if (TableBuiltinScalarFunction.BIT_COUNT
+        .getFunctionName()
+        .equalsIgnoreCase(functionName)) {
+      ColumnTransformer first = this.process(children.get(0), context);
+      if (isLongLiteral(children.get(1))) {
+        final long bits = ((LongLiteral) children.get(1)).getParsedValue();
+        return new BitCountColumnTransformer(INT64, first, bits);
+      } else {
+        return new BitCount2ColumnTransformer(INT64, first, this.process(children.get(1), context));
+      }
+    } else if (TableBuiltinScalarFunction.BITWISE_AND
+            .getFunctionName()
+            .equalsIgnoreCase(functionName)
+        || TableBuiltinScalarFunction.BITWISE_NOT.getFunctionName().equalsIgnoreCase(functionName)
+        || TableBuiltinScalarFunction.BITWISE_OR.getFunctionName().equalsIgnoreCase(functionName)
+        || TableBuiltinScalarFunction.BITWISE_XOR
+            .getFunctionName()
+            .equalsIgnoreCase(functionName)) {
+      ColumnTransformer first = this.process(children.get(0), context);
+      if (children.size() == 1) {
+        return new BitwiseColumnTransformer(INT64, first);
+      } else if (children.size() == 2) {
+        if (isLongLiteral(children.get(1))) {
+          final long rightValue = ((LongLiteral) children.get(1)).getParsedValue();
+          return new BitwiseColumnTransformer(INT64, first, rightValue, functionName);
+        } else {
+          return new Bitwise2ColumnTransformer(
+              INT64, first, this.process(children.get(1), context), functionName);
+        }
+      }
+    } else if (TableBuiltinScalarFunction.BITWISE_LEFT_SHIFT
+            .getFunctionName()
+            .equalsIgnoreCase(functionName)
+        || TableBuiltinScalarFunction.BITWISE_RIGHT_SHIFT
+            .getFunctionName()
+            .equalsIgnoreCase(functionName)
+        || TableBuiltinScalarFunction.BITWISE_RIGHT_SHIFT_ARITHMETIC
+            .getFunctionName()
+            .equalsIgnoreCase(functionName)) {
```

Comment:
implement different ColumnTransformer for each function, you can extract common logics in the super class, and only leave different process logic in each sub class.